Document stl error exceptions. (#8)
stainless-api/stl-api: docs/docs/endpoints.md

or resolve to a raw response to be parsed by the `response` schema.

The `params`, `context` and return types will be automatically mapped from the request and response schemas.

## Error handling

> In order to indicate an unauthorized usage of or error encountered during the 
> handling of an endpoint, throw an exception within the `handler`.

Reviewer: ```suggestion
You can throw certain exceptions within the `handler` to return 4xx or 5xx HTTP error responses to the user, with well-formatted bodies.
```

Author: Kind of a side question, but do you see 3xx status codes as something we should support? The exception mechanism is probably not the right one, at least for some codes in the group.

Reviewer: My instinct is no, API routes should not typically 3xx, but I may be wrong about that. I can't immediately think of a case where this would make sense, so I'm not sure whether raising or returning to 3xx would make more sense.

`stl` provides some exception types to streamline this. Each will cause the endpoint to respond with an HTTP status code. 

> Each exception's constructor also optionally accepts an record
> parameter. If provided, this parameter is encoded in JSON and included
> as the body of the response.

Reviewer: ```suggestion
Each exception's constructor also accepts an optional record
parameter, which is included in the body of the response JSON.
```

### `stl.StlError`

Takes an HTTP status code as a constructor argument. Responds with 

> that status code.

Reviewer: ```suggestion
that status code. Best used to create custom subclasses if you need HTTP status codes we don't yet provide out of the box.
```
